modules/core/src/smithy4s/ShapeId.scala

```diff
@@ -24,6 +24,14 @@ final case class ShapeId(namespace: String, name: String) extends HasId {
 }
 
 object ShapeId extends ShapeTag.Has[ShapeId] {
+  def parse(string: String): Option[ShapeId] = {
```

Comment:
if we really need this, I suggest we implemente it on top of `ShapeId.from(s: String)` from smithy-model itself. You may need to catch `ShapeIdSyntaxException`